app/components/shared/ImageUploadField.js

```diff
@@ -213,37 +238,79 @@ export default class ImageUploadField extends React.PureComponent<Props, State>
         );
 
         if (imageFiles.length === 1) {
-          if (this.state.lastImageUrl) {
-            URL.revokeObjectURL(this.state.lastImageUrl);
+          const imageFile = imageFiles[0];
+
+          // If the image is less than 256 kb, is an svg, or is a gif less than 768kb, don't even bother resizing
+          if (imageFile.size <= 262144 ||
+              imageFile.type === 'image/svg' ||
+              (imageFile.type === 'image/gif' && imageFile.size <= 786432)) {
+            this.processUpload(imageFile);
+            return;
           }
 
-          this.setState({
-            uploading: true,
-            currentImageUrl: URL.createObjectURL(imageFiles[0]),
-            lastImageUrl: this.state.currentImageUrl
-          });
-
-          this.assetUploader.uploadFromArray(imageFiles);
-        } else if (imageFiles.length > 1) {
-          const error = new Error('Only one image can be uploaded.');
-          this.setState(
-            { error },
-            () => {
-              if (this.props.onError) {
-                this.props.onError(error);
+          this.withLoadImage(
+            (loadImage: Function) => loadImage(
+              imageFile,
+              (processed: HTMLImageElement | HTMLCanvasElement | Event) => {
+                if (processed instanceof HTMLCanvasElement) {
+                  processed.toBlob(
+                    (blob: Blob) => {
+                      // If we didn't improve things, then let's just upload the original
+                      if (blob.size >= imageFile.size) {
+                        this.processUpload(imageFile);
+                      } else {
+                        this.processUpload(blob);
+                      }
+                    },
+                    imageFile.type
+                  );
+                } else {
+                  // If loadImage gives us an <img/>, or otherwise couldn't resize it, fall back
+                  this.processUpload(imageFile);
+                }
+              },
+              {
+                minHeight: 50,
+                minWidth: 50,
+                maxHeight: 500,
```

Comment:
My only question now is: is 500px enough?? It gives us max 250px on a @2x display, or 166px if a @3x ever comes into existence. I'd say that's probably okay? We could make it 1000px if we wanted to be safe 🤔 What do you reckon? I really don't know.

Reply:
We should totally up the mins though! I only just noticed they're only 50px.

Reply:
GitHub’s icons are only 200px (shown at 100pt) at retina, for reference. Depends where you want to take it in the future! I’d suggest that if we want to make it much bigger than 500px we might need to consider some sort of remote image resize solution like IIRC AWS offer.

(Please ignore the min values; that just means if you upload a thing between 50 and 500px we won’t scale it up (this resize utility is pretty bad it turns out))

Reply:
actually now I look at it the min should probably be, like, `minWidth: 1` tbqh, we don’t ever want to scale upward